### DISCUSSION TITLE
Login fails when using dnsmasq proxy

### DISCUSSION THREAD

**Discussion by wfragakis** [ltsp/ltsp#882] [General] [answered]
Hi all,
Have been happily using the current version of LTSP  on Debian for four years using a separate isc-dhcp-server . With this package having gone EOL, it's time to make a change. We already have a gateway capable of running dhcp so I thought I'd make my life simple and use the dnsmasq proxy feature on the LTSP server. For my testing, I'm using the same dhcp server that works fine when not using dnsmasq proxy.

Testing on a VM, the client boots fine but login fails. The only way to log in is to include the credentials as a auto login from the ltsp.conf file. Otherwise, either in the GUI (F7) or terminal (F1), I get .Login incorrect.

Relevant ltsp-dnsmasq.conf changes:
dhcp-range=set:proxy,192.168.0.0,proxy,255.255.255.0
dhcp-option=option:dns-server,192.168.0.1,1.1.1.1,1.0.0.1

Am I missing something obvious? Thanks!

**Comment by alkisg** [comment]
Boot with your **old** setup and run `ip a`, `ip r` and `ping server` on the client. To get a shell, you can either use the autologin that works for you, or login as root via ltsp.conf or via epoptes.
Then boot with the **new** setup, run the same commands, and compare. They should be the same.

**Comment by wfragakis** [comment]
Thanks, the command outputs are (not -see below) identical. Yet, if I do not include the credentials in ltsp.conf, I can't log in. I can only log in via dnsmasq proxy if I include the login credentials on ltsp.conf instead of the normal behaviour of accounts on the ltsp server, /etc/passwd, /etc/shadow, etc (and remote server for /home).

Edit: The output of ping server is not identical. Both pings worked but I missed that they were pinging different addresses

Because this was built on an old legacy ltps thin client install, "server" was the old ltsp server/dhcp server (e.g 192.168.0.254) and that host file has been replicated across our network.

In the proxy version, "ping server" pings .254 ie the dhcp server
In the non-proxy working version, it pings .15 ie the ltsp tftp server 
SERVER is defined in ltsp.conf as SERVER=192.168.0.15 under the [SERVER] section

**Comment by wfragakis** [comment, marked as answer]
Well, my own reply suggested the answer. Moving the SERVER directive to the [COMMON] section solved the issue.

Thanks for the ideas that prompted my looking at that. I would have been lost for hours if not days without your help.
